### PR TITLE
Chore: Clean up intervalv2 functions

### PR DESCRIPTION
**Pull request by itsmylife**
**What is this feature?**

**TLDR**
A lot of import path changes/updates.

**Detailed**
For the decoupling effort of different datasources, there are duplicated code pieces of `intervalv2` package. 

With the following PRs, we now have most of its functionality in grafana-plugin-sdk-go
https://github.com/grafana/grafana-plugin-sdk-go/pull/880
https://github.com/grafana/grafana-plugin-sdk-go/pull/886

So there is no reason to keep the duplicated code. We are duplicating the `intervalv2.Calculator`. Because different datasources need different parts of it and I am not just ready to make a big change. 

**What else needs to be done**
- [x] https://github.com/grafana/grafana/pull/82064 Remove from prometheus 
- [x] https://github.com/grafana/grafana-plugin-sdk-go/pull/887 Waiting it to be merged so I will handle `roundInterval` function too.

**Why do we need this feature?**

Less code duplication

**Who is this feature for?**

AzureMonitor, Cloud-Monitoring, PublicDashboards and Loki

### What has been done?
- Functions that was moved to `grafana-plugin-sdk-go` removed from `intervalv2`
- Places that depends on `intervalv2` updated
  - Functions are being used from `grafana-plugin-sdk-go`
  - Calculator is copied to places where it's needed. It has been done that way so we decoupled it
    -  In the future I am planning to unify it (as far as I see we all use it as it is, but that needs a bit more careful investigation)
- Original `intervalv2` is removed.

### CODE REVIEW

**Review by ivanahuckova — APPROVED** [review]
LGTM for Loki! Thank you @itsmylife a lot for updating this inLoki data source. Super appreciated! 🌷

File: pkg/tsdb/legacydata/interval/interval.go [hunk 1]
```diff
@@ -117,126 +116,11 @@ func GetIntervalFrom(dsInfo *datasources.DataSource, queryModel *simplejson.Json
...
-	// 3w
-	case interval <= 1814400000*time.Millisecond:
-		return time.Millisecond * 604800000 // 1w
-	// 2y
-	case interval < 3628800000*time.Millisecond:
-		return time.Millisecond * 2592000000 // 30d
-	default:
-		return time.Millisecond * 31536000000 // 1y
+	if interval <= 15*time.Millisecond {
+		return time.Millisecond * 10
```
Comment (itsmylife) [inline pkg/tsdb/legacydata/interval/interval.go:123]: In `grafana-plugin-sdk-go/backend/gtime` there is `0.01s` precision. https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/gtime/gtime.go#L175-L177
That's why I do a special check here. Maybe legacydata doesn't have such interval. So I don't want to change behavior while removing duplicate code.

File: pkg/tsdb/legacydata/interval/interval.go [hunk 2]
```diff
@@ -117,126 +116,11 @@ func GetIntervalFrom(dsInfo *datasources.DataSource, queryModel *simplejson.Json
 	return parsedInterval, nil
 }
 
-// FormatDuration converts a duration into the kbn format e.g. 1m 2h or 3d
-func FormatDuration(inter time.Duration) string {
```
Comment (itsmylife) [inline pkg/tsdb/legacydata/interval/interval.go:121]: Same code https://github.com/grafana/grafana-plugin-sdk-go/blob/main/backend/gtime/gtime.go#L90-L120

**Review by kylebrandt — APPROVED** [review]

**Review by evictorero — APPROVED** [review]

File: pkg/tsdb/legacydata/interval/interval.go [hunk 3]
```diff
@@ -117,126 +114,11 @@ func GetIntervalFrom(dsInfo *datasources.DataSource, queryModel *simplejson.Json
...
-	// 3w
-	case interval <= 1814400000*time.Millisecond:
-		return time.Millisecond * 604800000 // 1w
-	// 2y
-	case interval < 3628800000*time.Millisecond:
-		return time.Millisecond * 2592000000 // 30d
-	default:
-		return time.Millisecond * 31536000000 // 1y
+	if interval <= 15*time.Millisecond {
+		return time.Millisecond * 10
```
Comment (suntala) [inline pkg/tsdb/legacydata/interval/interval.go:121]: nit:
```suggestion
		return time.Millisecond * 10 // 0.01s
```

**Review by suntala — APPROVED** [review]
